app/code/Magento/CatalogUrlRewrite/Observer/ProductProcessUrlRewriteSavingObserver.php

```diff
@@ -44,32 +49,29 @@ public function __construct(
     ) {
         $this->productUrlRewriteGenerator = $productUrlRewriteGenerator;
         $this->urlPersist = $urlPersist;
-        $this->productUrlPathGenerator = $productUrlPathGenerator ?: ObjectManager::getInstance()
-            ->get(ProductUrlPathGenerator::class);
+        $this->productUrlPathGenerator = $productUrlPathGenerator
+            ?: ObjectManager::getInstance()->get(ProductUrlPathGenerator::class);
```

Comment:
backward compatibility policy is not needed for observers, I think we can make this argument not optional